iceoryx_posh/test/moduletests/test_posh_runtime.cpp

```diff
@@ -148,12 +148,23 @@ TEST_F(PoshRuntime_test, NoAppName)
 }
 
 // To be able to test the singleton and avoid return the exisiting instance, we don't use the test fixture
-TEST(PoshRuntime, LeadingSlashAppNameLeadsToFailure)
+TEST(PoshRuntime, RuntimeFailsWhenAppNameIsNotAFileName)
 {
     ::testing::Test::RecordProperty("TEST_ID", "77542d11-6230-4c1e-94b2-6cf3b8fa9c6e");
-    const iox::RuntimeName_t invalidAppName = "/miau";
 
-    EXPECT_DEATH({ PoshRuntime::initRuntime(invalidAppName); }, "POSH__RUNTIME_LEADING_SLASH_PROVIDED");
+    for (auto i : {"/miau",
+                   "/fuu/bar",
+                   "plum/bus",
+                   ".",
+                   "..",
+                   "strawberriesWithMayonnaiseIs/..",
+                   "ohLookADot.",
+                   "amIADirectory/"})
```

Comment:
I think you missed `IAmASpace ` and the empty case. In the end we're just duplicating the tests from `test_cxx_helplets.cpp`. Maybe this is not needed and we just test with one case because we know that the underlying `isValidFileName` works as expected?

Reply:
@elfenpiff Ok. However, if we now support a new file system or platform we have to adapt two tests at the same time and one might forget either this one here or the on in helplets. Maybe we can share the test cases? It's not a biggie and can be solved on another PR.